scripts/microbe_stage/patch_system.lua

```diff
@@ -0,0 +1,90 @@
+--------------------------------------------------------------------------------
+-- PatchComponent
+-- 
+-- Controls population management within a certain region of space
+--------------------------------------------------------------------------------
+
+class "PatchComponent" (Component)
+
+function PatchComponent:__init()
+	Component.__init(self)
+	-- stuff
+end
+
+--[[
+We need a bunch of stuff handled here:
+Compound movement:
+- environment -> species
+- species -> environment
+- predation: species -> other species and environment
+
+Population changes:
+- immigration -- add tentative populations, keep those viable enough to displace 
+	another species (and give them resources of species displaced, with leakage)
+- extinction, by culling
+- speciation? We may need to slightly control here when it happens
+
+Culling:
+- when population drops below sustainable level, free all compounds
+- when population mutates, and criteria for branching not met, move compounds 
+	from ancestor to new, with leakage
+- another case?
+
+--]]
```

Comment:
yes exactly, I'd continue doing this work but there is a lot of refactoring on this branch which it would be better to merge